### PR TITLE
Add should - testing library with zero dependencies

### PR DESCRIPTION
**Pull request by Andrei-hub11**
## We want to ensure high quality of the packages. Make sure that you've checked the boxes below before sending a pull request.

- [x] I have read the [Contribution Guidelines](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#contribution-guidelines)
- [x] I have read the [Maintainers Note](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#maintainers)
- [x] I have read the [Quality Standards](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#quality-standards)

_Not every repository (project) will require every option, but most projects should. Check the Contribution Guidelines for details._

- [x] The repo documentation has a pkg.go.dev link.
- [x] The repo documentation has a coverage service link.
- [x] The repo documentation has a goreportcard link.
- [x] The repo has a version-numbered release and a go.mod file.
- [x] The repo has a continuous integration process that automatically runs tests that must pass before new pull requests are merged.
- [x] Continuous integration is used to attempt to catch issues prior to releasing this package to end-users.

## Please provide some links to your package to ease the review

- [x] forge link (github.com, gitlab.com, etc): https://github.com/Kairum-Labs/should
- [x] pkg.go.dev: https://pkg.go.dev/github.com/Kairum-Labs/should
- [x] goreportcard.com: https://goreportcard.com/report/github.com/Kairum-Labs/should
- [x] coverage service link ([codecov](https://codecov.io/), [coveralls](https://coveralls.io/), etc.): https://app.codecov.io/gh/Kairum-Labs/should

## Pull Request content

- [x] The package has been added to the list in alphabetical order.
- [x] The package has an appropriate description with correct grammar.
- [x] As far as I know, the package has not been listed here before.

## Category quality

_Note that new categories can be added only when there are 3 packages or more._

Packages added a long time ago might not meet the current guidelines anymore. It would be very helpful if you could check 3-5 packages above and below your submission to ensure that they also still meet the Quality Standards.

Please delete one of the following lines:

- [x] The packages around my addition still meet the Quality Standards.

Thanks for your PR, you're awesome! :sunglasses:

## Summary by CodeRabbit

* **Documentation**
  * Added a new testing library entry "should" (description and link) under the Testing Frameworks section in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1916,7 +1916,7 @@ _These libraries were placed here because none of the other categories seemed to
 - [xdg](https://github.com/rkoesters/xdg) - FreeDesktop.org (xdg) Specs implemented in Go.
 - [xkg](https://github.com/go-xkg/xkg) - X Keyboard Grabber.
 - [xz](https://github.com/ulikunitz/xz) - Pure golang package for reading and writing xz-compressed files.
-**[⬆ back to top](#contents)**
+  **[⬆ back to top](#contents)**
```
Comment (ccoVeille) [inline README.md]: I'm unsure why you changed this

File: README.md [hunk 2]
```diff
@@ -2090,7 +2090,7 @@ _Libraries for making HTTP requests._
 - [go-req](https://github.com/wenerme/go-req) - Declarative golang HTTP client.
 - [go-retryablehttp](https://github.com/hashicorp/go-retryablehttp) - Retryable HTTP client in Go.
 - [go-zoox/fetch](https://github.com/go-zoox/fetch) - A Powerful, Lightweight, Easy Http Client, inspired by Web Fetch API.
-- [Grequest](https://github.com/lib4u/grequest)  - Simple and lightweight golang package for http requests. based on powerful net/http
+- [Grequest](https://github.com/lib4u/grequest) - Simple and lightweight golang package for http requests. based on powerful net/http
```
Comment (ccoVeille) [inline README.md]: Don't update line you don't need.

I assume you have an IDE that strips trailing spaces.

But examine your commit line by line

Comment (Andrei-hub11) [reply README.md]: Yeah, it was the IDE. I fixed it

**Comment by Andrei-hub11** [comment]
@avelino, can you took a look, please?

**Comment by Andrei-hub11** [comment]
@phanirithvij, can you took a look, please?

**Comment by geralxtra** [comment]
Show me ,what do you mean "zero dependencies"

**Comment by Andrei-hub11** [comment]
> Show me ,what do you mean "zero dependencies"

What I meant by “zero dependencies” is that the package relies exclusively on Go’s standard library. This reduces risks introduced by third-party packages, simplifies the dependency graph, and makes it easier to adopt should in projects that prioritize lightweight and stable modules.

**Comment by geralxtra** [comment]
Lowers risks and modifies modules right..
